interpreter/syntax/ast.ml

```diff
@@ -62,10 +62,14 @@ type 'a memop =
   {ty : value_type; align : int; offset : Memory.offset; sz : 'a option}
 type loadop = (Memory.pack_size * Memory.extension) memop
 type storeop = Memory.pack_size memop
-type atomicloadop = storeop (* Memory.extension isn't used. *)
-type atomicstoreop = storeop
-type atomicrmwop = rmwop * storeop
-type atomicrmwcmpxchgop = storeop
+
+type atomicmemop = storeop (* No atomic operations use Memory.extension *)
```

Comment:
I would perhaps remove all the aliases to this type (and rename this to just `atomicop`?); `atomicrmwop` could go as well -- it seems more logical to have the rmwop as a separate argument directly on the `AtomicRMW` constructor, since it isn't really related to the memory access per se.

Reply:
Done